tests/keyboard/index.ts

```diff
@@ -163,3 +163,30 @@ test('disabling activeElement moves action to HTMLBodyElement', async () => {
     body - keyup: c
   `)
 })
+
+test('typing on focused element with shadow root', async () => {
+  const keypressHandler = mocks.fn()
+  class CustomElement extends HTMLElement {
+    constructor() {
+      super()
+
+      this.attachShadow({mode: 'open'})
+      this.addEventListener('keypress', keypressHandler)
+    }
+  }
+
+  customElements.define('custom-element', CustomElement)
```

Comment:
The element definition should be moved out of the test function to the bottom of the test file because the test should have no side-effect and the custom element will be defined for any tests that might be added below.

Naming it e.g. `focusable-shadow-root` might help someone working on this file and I think inlining the class definition into the `define()` might help if someone decides to refactor the file later.

Reply:
The custom element isn't actually focusable by itself, but we add the `tabindex` attribute to it in the test to make it focusable. I'm going to (conditionally for best practice) add a tabindex in the constructor so that this custom element serves this single purpose. Does that make sense?

Reply:
Like this:

```suggestion
  customElements.define(
    'focusable-custom-element',
    class FocusableCustomElement extends HTMLElement {
      constructor() {
        super()
        this.attachShadow({mode: 'open'})
      }
    
      connectedCallback() {
        if (!this.hasAttribute('tabindex')) {
          this.setAttribute('tabindex', '0')
        }
      }
    }
  )
```

Reply:
I think either would be okay:
`<custom-element tabindex="0"></custom-element>` or `<focusable-custom-element></focusable-custom-element>`.

Just move it out of the test.